stream-tests/src/test/scala/org/apache/pekko/stream/scaladsl/HubSpec.scala

```diff
@@ -561,6 +561,49 @@ class HubSpec extends StreamSpec {
       out.expectComplete()
     }
 
+    "handle unregistration concurrent with registration" in {
+
+      var sinkProbe1: TestSubscriber.Probe[Int] = null
+
+      def registerConsumerCallback(id: Long): Unit = {
+        if (id == 1) {
+          sinkProbe1.cancel()
+          Thread.sleep(10)
```

Comment:
is this sleep useful?

Reply:
It was probably for my local test case in earlier versions. I can now reproduce the broken behavior without the sleep on the old implementation. I'll remove it.